pkg/reconciler/ingress/ingress_test.go

```diff
@@ -247,61 +252,85 @@ func withAnnotations(ans map[string]string) ingressCreationOption {
 	}
 }
 
-func withPreferHeaderPaths(isAlwaysAsync bool) ingressCreationOption {
-	return func(ing *v1alpha1.Ingress) {
-		// Get ingress name depending on async property
-		var ingName string
-		var ingNamespace string
-		if isAlwaysAsync {
-			ingName = ingAlwaysAsync.Name
-			ingNamespace = ingAlwaysAsync.Namespace
-		} else {
-			ingName = ingWithAsyncAnnotation.Name
-			ingNamespace = ingWithAsyncAnnotation.Namespace
-		}
-		splits := make([]v1alpha1.IngressBackendSplit, 0, 1)
-		splits = append(splits, v1alpha1.IngressBackendSplit{
-			IngressBackend: v1alpha1.IngressBackend{
-				ServiceName:      ingName + asyncSuffix,
-				ServiceNamespace: ingNamespace,
-				ServicePort:      intstr.FromInt(80),
+func expectedCreatedIng(namespace, name string, status v1alpha1.IngressStatus, isAlwaysAsync bool) *v1alpha1.Ingress {
```

Comment:
instead of isAlwaysAsync what if we pass the paths in? that way you can have an ingressWithPaths() method that wont have any branches in it. `createdIng` then becomes `createdIng = ingressWithPaths(blah, blah, []netv1alpha1.HTTPIngressPath{{ .. }})`. (Just trying to drop a bit more same-as-the-code-being-tested logic from the tests).

Reply:
makes sense, just updated!